### PR TITLE
Upgrade PMD to 6.3.0

### PR DESCRIPTION
**Pull request by doncem**
Long awaited update. Using categories

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/demo/controllers/GetWelcomeTest.java [hunk 1]
```diff
@@ -17,7 +17,7 @@
 public class GetWelcomeTest {
 
     @Autowired
-    private MockMvc mockMvc;
+    private transient MockMvc mockMvc;
```
Comment (lgonczar) [inline src/test/java/uk/gov/hmcts/reform/demo/controllers/GetWelcomeTest.java]: Is `transient` required here?

File: config/pmd/ruleset.xml [hunk 2]
```diff
@@ -4,31 +4,40 @@
   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
   xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 http://pmd.sourceforge.net/ruleset_2_0_0.xsd">
 
-  <description>
-    HMCTS PMD rule set
-  </description>
+  <description>HMCTS PMD rule set</description>
```
Comment (lgonczar) [inline config/pmd/ruleset.xml]: Is there a plan to update other java projects with this change?

File: src/test/java/uk/gov/hmcts/reform/demo/controllers/GetWelcomeTest.java (hunk 1 above)
Comment (doncem) [reply src/test/java/uk/gov/hmcts/reform/demo/controllers/GetWelcomeTest.java]: fixes error prone [rule](https://pmd.github.io/pmd-6.3.0/pmd_rules_java_errorprone.html#beanmembersshouldserialize)

File: config/pmd/ruleset.xml (hunk 2 above)
Comment (doncem) [reply config/pmd/ruleset.xml]: We should update pmd tool. Better rule tuning definitely will happen in bigger projects and those can be summarised over here

File: src/test/java/uk/gov/hmcts/reform/demo/controllers/GetWelcomeTest.java (hunk 1 above)
Comment (lgonczar) [reply src/test/java/uk/gov/hmcts/reform/demo/controllers/GetWelcomeTest.java]: 👍

**Pull request merged**